src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -741,13 +757,37 @@ void medViewContainer::openFromSystem()
     if (path.isEmpty())
         return;
 
-    connect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)), this, SLOT(dataReady(medDataIndex,QUuid)));
-    d->expectedUuid = medDataManager::instance()->importPath(path, true, false);
+    open(path);
+}
 
+void medViewContainer::open(const QString & path)
+{
+    QEventLoop loop;
+    QUuid uuid = medDataManager::instance()->importPath(path, false);
+    d->expectedUuids.append(uuid);
+    connect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)),
+            this, SLOT(open_waitForImportedSignal(medDataIndex,QUuid)));
+    while( d->expectedUuids.contains(uuid)) {
+        loop.processEvents(QEventLoop::ExcludeUserInputEvents);
+    }
     //  save last directory opened in settings.
     medSettingsManager::instance()->setValue("path", "medViewContainer", path);
 }
 
+
+void medViewContainer::open_waitForImportedSignal(medDataIndex index, QUuid uuid)
+{
```

Comment:
I did that in case secureImport is called by multiple threads. In practice we don't do that, but if someday a pipeline tries to do two imports at the same time, it won't bug. It doesn't cost anything to have it there.
The important point about the code I posted wasn't so much the mutexes (since currentExports isn't used as a loop condition like expectedUuids is) but the use of signals to end the loop.